pkg/services/publicdashboards/service/service.go

```diff
@@ -167,23 +172,31 @@ func (pd *PublicDashboardServiceImpl) updatePublicDashboardConfig(ctx context.Co
 
 // BuildPublicDashboardMetricRequest merges public dashboard parameters with
 // dashboard and returns a metrics request to be sent to query backend
-func (pd *PublicDashboardServiceImpl) BuildPublicDashboardMetricRequest(ctx context.Context, dashboard *models.Dashboard, publicDashboard *PublicDashboard, panelId int64) (dtos.MetricRequest, error) {
+func (pd *PublicDashboardServiceImpl) BuildPublicDashboardMetricRequest(ctx context.Context, dashboard *models.Dashboard, publicDashboard *PublicDashboard, panelId int64, reqDTO *PublicDashboardQueryDTO) (dtos.MetricRequest, error) {
 	if !publicDashboard.IsEnabled {
```

Comment:
This code is only called from the api which calls GetDashboard(accessToken) and returns ErrPublicDashboardNotFound so this code would not actually get hit. I'm wondering if for clarity, it's worth removing the isEnabled check in this method now that we have the additional complexity of the PublicDashboardQueryInterval

Reply:
Agree, we shouldn't call this method if the pubdash is disabled. Moved this check to the upper layer. I don't think this struct implies we have the public dashboard info because it is a DTO, it only implies we receive a request with data from a client.